reference/extensions/binary_compatibility.rst

```diff
@@ -0,0 +1,24 @@
+.. _reference_extensions_binary_compatibility:
+
+Binary compatibility
+--------------------
+
+This plugin, located in the cache ``extensions/plugins/compatibility/compatibility.py`` allows defining custom
+rules for the binary compatibility of packages accross settings and options.
+It has some built-in logic implemented, but can be customized.
+
+
+.. code-block:: python
+
+    def compatibility(conanfile):
+        result = []
+        if conanfile.settings.build_type == "Debug":
+            result.append({"settings": [("build_type", "Release")]})
+        return result
+  
+
+Some important rules:
+
+- The built-in ``compatibility.py`` is subject to changes in future releases. To avoid being updated
+  in the future, please remove the first comment ``# This file was generated by Conan.``
```

Comment:
this looks suspicious. comments should not influence on the behavior, it's counter-intuitive and breaks user-expectations (well, there are special comments in python to control encoding, for instance, but they have specific syntax, this example looks just as a regular comment).
if we want to prevent auto-update, why not use something more explicit (an attribute, or an entry in conf?)

Reply:
There are a bunch of files that are generated by Conan. Having a comment in them is very explicit, easy to read and understand when you open the files. Having a conf for every different file will be hidden and more complex.